### ISSUE TITLE
Config NPM_DEPENDENCIES vs. SYSTEM_CONFIG_DEV vs. SYSTEM_BUILDER_CONFIG

### ISSUE THREAD

**Issue by brian428** [mgechev/angular2-seed#1364]
Minko, I was going to update this wiki page (https://github.com/mgechev/angular2-seed/wiki/Add-PrimeNG) to replace the step that involves editing seed.config.ts, so edits are made to project.config.ts instead.

My first instinct was to simply change it add the library to `NPM_DEPENDENCIES` in project.config.ts. However, that made me realize that `NPM_DEPENDENCIES` might not be the place to do this.

For "plain old" external JavaScript libraries, `NPM_DEPENDENCIES` seems to make sense. However, for libraries like ngrx and primeng, I'm less certain.

I'm sure that the problem is my limited understanding of exactly how SystemJS plays into the build process (and possibly even the AoT compile process).

Would adding node modules like ngrx or primeng to `SYSTEM_CONFIG_DEV.packageConfigPaths` and `SYSTEM_BUILDER_CONFIG.packages` be a better option? Would this allow for tree-shaking or other optimizations that aren't available if an external .js bundle file is just stuck into `NPM_DEPENDENCIES`?

Hopefully this makes sense. I just want to be sure I understand the impact of the different options before I go edit that wiki page (and maybe even add one covering ngrx).

**Comment by mgechev** [comment]
`NPM_DEPENDENCIES` is used in rare cases, mostly when you need to inject scripts which expose a global variable or there's a CSS file that you want to include in your project (`bootstrap.css` for instance).

`SYSTEM_CONFIG_DEV` is used for configuring SystemJS. In some cases the base configuration provided is enough, however, sometimes you may need to add custom mapping for your dependencies.

In the perfect world `SYSTEM_CONFIG_DEV ` and `SYSTEM_BUILDER_CONFIG` should be the same, however, sometimes we want to include a minified version of a dependency in our production build or the SystemJS builder requires slightly different configuration.

In short - almost all modules should reside in `SYSTEM_CONFIG_DEV` and `SYSTEM_BUILDER_CONFIG` except ones which expose global globals, then they should be in `NPM_DEPENDENCIES`.

**Comment by geminiyellow** [comment]
@mgechev should add this to wiki `best practice`.
